pommerman/network/ion_server/__init__.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+"""IonServer - This library allows playing of matches on network via the
+WebSockets protocol.  
+Functions:  
+init() - If you want to run the application normally  
+run(max_players, max_matches, port, timeout, mode, ui_en=True, 
+exit_handler=True) - If you want to programatically launch the server with
+predefined parameters"""
+
+import ui
+import multiprocessing
+from . import constants
+# This is for switching between languages incase there are more
+Strings = constants.LangEN
+from . import network
+from . import match
+import time
+import random
+import signal
+import pommerman
+
+MATCHSP = []
+
+
+def _exit_handler(sp_net):
+    """Description: Return the exit handler with a reference to the sp_net
+    variable."""
+
+    def exit_handler(_s, _h):
+        """Description: Handle exiting the application."""
+        ui.info(ui.yellow, "Exiting..")
+        sp_net.terminate()
+        for i in MATCHSP:
+            i.terminate()
+        exit(0)
+
+    return exit_handler
+
+
+def init():
+    """Description: Initiate the application by asking questions."""
+    ui.info(ui.yellow, Strings.sever_starting.value)
+    port = int(ui.ask_string(Strings.server_port.value))
+    max_players = int(ui.ask_string(Strings.server_maxp.value))
+    if max_players < 4:
+        ui.fatal(ui.yellow, "Input a number greater than 4")
+    modes = []
+    for i in pommerman.configs.__dir__():
+        if i[-4:] == "_env":
+            id = getattr(pommerman.configs, i)()["env_id"]
+            if id[-2:] != "v2":
+                modes.append(id)
+    run(max_players, int(max_players / 4), port,
+        float(ui.ask_string(Strings.server_to.value)),
+        str(ui.ask_choice(Strings.server_mode.value, modes)))
```

Comment:
Ask for these earlier, not in the function call. That way, both you can handle problems and we would know where the problem lies.

Reply:
Ditto.